src/Xamarin.Android.Build.Tasks/Microsoft.Android.Sdk/targets/Microsoft.Android.Sdk.BuildOrder.targets

```diff
@@ -45,18 +45,8 @@ projects, these properties are set in Xamarin.Android.Legacy.targets.
     </BuildDependsOn>
   </PropertyGroup>
 
-  <PropertyGroup>
```

Comment:
Why did this property group need to be moved to be after `$(CoreResolveReferencesDependsOn)` and `$(ResolveReferencesDependsOn)`?

Reply:
I needed to add a `Condition` so that `ResolveSdks` didn't run in the nested `<MSBuild/>` call:

> _ComputeFilesToPublishForRuntimeIdentifiers was running some extra
targets causing <ResolveSdks/> to run 5 times for 4 ABIs. I had to
make changes in BuildOrder.targets to fix this.

Reply:
I still don't comprehend this answer.  The `Condition`al `<PropertyGroup/>` covers `$(CoreResolveReferencesDependsOn)` and `$(ResolveReferencesDependsOn)`, which doesn't *immediately* appear to cover/include `$(CompileDependsOn)` or `$(CoreCompileDependsOn)`.

Rephrased, I fail to see how *this* wouldn't work, which would result in a smaller diff (I think?):

```xml
  <PropertyGroup>
    <CompileDependsOn>
      _SetupDesignTimeBuildForCompile;
      _AddAndroidDefines;
      _IncludeLayoutBindingSources;
      AddLibraryJarsToBind;
      $(CompileDependsOn);
    </CompileDependsOn>
    <CoreCompileDependsOn>
      UpdateGeneratedFiles;
      $(CoreCompileDependsOn)
    </CoreCompileDependsOn>
  </PropertyGroup>

  <!-- Targets that run unless we are running _ComputeFilesToPublishForRuntimeIdentifiers -->
  <PropertyGroup Condition=" '$(_ComputeFilesToPublishForRuntimeIdentifiers)' != 'true' ">
    <CoreResolveReferencesDependsOn>
      _SeparateAppExtensionReferences;
      _PrepareWearApplication;
      $(ResolveReferencesDependsOn);
      _AddAndroidCustomMetaData;
    </CoreResolveReferencesDependsOn>
    <ResolveReferencesDependsOn>
      $(CoreResolveReferencesDependsOn);
      UpdateAndroidInterfaceProxies;
      UpdateAndroidResources;
      AddBindingsToCompile;
      AddEmbeddedJarsAsResources;
      AddEmbeddedReferenceJarsAsResources;
    </ResolveReferencesDependsOn>
  </PropertyGroup>

  <PropertyGroup>
    <DeferredBuildDependsOn>
      $(DeferredBuildDependsOn);
      UpdateAndroidResources;
    </DeferredBuildDependsOn>
    <PrepareForRunDependsOn>
      $(PrepareForRunDependsOn);
      _CollectMonoAndroidOutputs;
    </PrepareForRunDependsOn>
    <CleanDependsOn>
      $(CleanDependsOn);
      _CleanMonoAndroidIntermediateDir;
      _CleanAndroidBuildPropertiesCache;
      CleanBindingsOutput;
      CleanLibraryProjectIntermediaries;
      CleanNativeLibraryIntermediaries;
    </CleanDependsOn>
    <ExportJarToXmlDependsOnTargets>
      _ResolveMonoAndroidSdks;
      _ExtractLibraryProjectImports;
      _GetLibraryImports;
      _ExportJarToXml;
    </ExportJarToXmlDependsOnTargets>
  </PropertyGroup>
```

…though I suppose the "problem" with ^^ is that it looks weird; why separate two non-`Condition`al property groups like that?  (The answer being "smaller patches!", but that might not actually be a good rationale…)

---

My question was *not* "why did `$(CoreResolveReferencesDependsOn)` and `$(ResolveReferencesDependsOn)` need to become conditional?".

My question was "why is this diff so noisy?"

Reply:
>  I suppose the "problem" with ^^ is that it looks weird; why separate two non-Conditional property groups like that? (The answer being "smaller patches!", but that might not actually be a good rationale…)

I think it should be ok to move code around that is being changed so the final code looks better? This would be like creating a new C# method when it gets too long. Diff size isn't a huge concern to me if it was code that is already changing.

If this was auto-formatting unrelated files, we shouldn't do that, of course.